src/inc/til/u8u16convert.h

```diff
@@ -76,32 +96,148 @@ namespace til // Terminal Implementation Library. Also: "Today I Learned"
-                if ((*(backIter - 1) & _Utf8BitMasks::MaskAsciiByte) > _Utf8BitMasks::IsAsciiByte)
+                if (codepage == gsl::narrow_cast<unsigned int>(CP_UTF8)) // UTF-8
                 {
-                    // Check only up to 3 last bytes, if no Lead Byte was found then the byte before must be the Lead Byte and no partials are in the string
-                    const size_t stopLen{ std::min(_buffer.length(), gsl::narrow_cast<size_t>(3u)) };
-                    for (size_t sequenceLen{ 1u }; sequenceLen <= stopLen; ++sequenceLen)
+                    auto backIter = _buffer.end();
+                    // If the last byte in the string was a byte belonging to a UTF-8 multi-byte character
+                    if ((*(backIter - 1) & _Utf8BitMasks::MaskAsciiByte) > _Utf8BitMasks::IsAsciiByte)
                     {
-                        --backIter;
-                        // If Lead Byte found
-                        if ((*backIter & _Utf8BitMasks::MaskContinuationByte) > _Utf8BitMasks::IsContinuationByte)
+                        // Check only up to 3 last bytes, if no Lead Byte was found then the byte before must be the Lead Byte and no partials are in the string
+                        const size_t stopLen{ std::min(_buffer.length(), gsl::narrow_cast<size_t>(3u)) };
+                        for (size_t sequenceLen{ 1u }; sequenceLen <= stopLen; ++sequenceLen)
                         {
-                            // If the Lead Byte indicates that the last bytes in the string is a partial UTF-8 code point then cache them:
-                            //  Use the bitmask at index `sequenceLen`. Compare the result with the operand having the same index. If they
-                            //  are not equal then the sequence has to be cached because it is a partial code point. Otherwise the
-                            //  sequence is a complete UTF-8 code point and the whole string is ready for the conversion into a UTF-16 string.
-                            if ((*backIter & _cmpMasks.at(sequenceLen)) != _cmpOperands.at(sequenceLen))
+                            --backIter;
+                            // If Lead Byte found
+                            if ((*backIter & _Utf8BitMasks::MaskContinuationByte) > _Utf8BitMasks::IsContinuationByte)
+                            {
+                                // If the Lead Byte indicates that the last bytes in the string is a partial UTF-8 code point then cache them:
+                                //  Use the bitmask at index `sequenceLen`. Compare the result with the operand having the same index. If they
+                                //  are not equal then the sequence has to be cached because it is a partial code point. Otherwise the
+                                //  sequence is a complete UTF-8 code point and the whole string is ready for the conversion to hstring.

```

Comment:
hstring? No, please continue to use UTF-16 string as we're going to use this in places in the code that have nothing to do with winrt/hstring.

Reply:
Haha, no hstring in the code, only in the comment. I don't recall where I copied it. Apparently I failed to update it properly though.